components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/scriptViewTabList.jsp

```diff
@@ -305,7 +306,7 @@
                              <% if (schemaList!= null){%>
                              <input class="button" type="button" value="< <fmt:message key="back"/>"  onclick="javascript:location.href = 'scriptViewSchemas.jsp?ordinal=1&tableList='+get_check_tableList()+'&flag=back';" />
            					 <% } else { %>
-           					 <input class="button" type="button" value="< <fmt:message key="back"/>" onclick="location.href = 'scriptAddSource.jsp?ordinal=1&flag=back&datasource=<%=sourceId%>&dbName=<%=dbName%>' "/>
+           					 <input class="button" type="button" value="< <fmt:message key="back"/>" onclick="location.href = 'scriptAddSource.jsp?ordinal=1&flag=back&datasource=<%=Encode.forHtmlAttribute(sourceId)%>&dbName=<%=Encode.forHtmlAttribute(dbName)%>' "/>
```

Comment:
forUriComponent should be used in both places.
https://owasp.github.io/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forUriComponent-java.lang.String